tests/functional/test_config.py

```diff
@@ -0,0 +1,488 @@
+            assert_action_has_issues(entity)
+    else:
+        action_status = entity.action_run(name='job').wait()
+        assert action_status == 'success'
+
+
+ASSERT_TYPE = {
+    'list': assert_list_type,
+    'map': assert_map_type,
+    'string': assert_string_type,
+    'password': assert_password_type,
+    'text': assert_text_type,
+    'file': assert_file_type,
+    'structure': assert_structure_type,
+    'boolean': assert_boolean_type,
+    'integer': assert_integer_type,
+    'float': assert_float_type,
+    'option': assert_option_type,
+}
+
+
+def assert_config_type(path, config_type, entities, is_required, is_default, sent_value_type):
+    for entity in entities:
+        ASSERT_TYPE[config_type](
+            path, config_type, entity, is_required, is_default, sent_value_type)
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'with_default', 'sent_correct_value', scope='module')
+def nr_wd_cv(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_with_default_sent_correct_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'with_default', 'sent_empty_value', scope='module')
+def nr_wd_ev(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_with_default_sent_empty_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'with_default', 'sent_null_value', scope='module')
+def nr_wd_nv(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_with_default_sent_null_value')
+
+
+def test_not_required_with_default_sent_correct_value(nr_wd_cv):
+    assert_config_type(*nr_wd_cv, False, True, 'correct_value')
+
+
+def test_not_required_with_default_sent_empty_value(nr_wd_ev):
+    assert_config_type(*nr_wd_ev, False, True, 'empty_value')
+
+
+def test_not_required_with_default_sent_null_value(nr_wd_nv):
+    assert_config_type(*nr_wd_nv, False, True, 'null_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'without_default', 'sent_correct_value', scope='module')
+def nr_wod_cv(sdk_client_ms: ADCMClient, request):
+    return processing_data(
+        sdk_client_ms, request, 'not_required_without_default_sent_correct_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'without_default', 'sent_empty_value', scope='module')
+def nr_wod_ev(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_without_default_sent_empty_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'without_default', 'sent_null_value', scope='module')
+def nr_wod_nv(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_without_default_sent_null_value')
+
+
+def test_not_required_without_default_sent_correct_value(nr_wod_cv):
```

Comment:
Test scenario should be commented here in help message form.

Reply:
Ok